internal/api/logs/v1/http.go

```diff
@@ -0,0 +1,145 @@
+package http
+
+import (
+    "net"
+    "net/http"
+    "net/http/httputil"
+    "net/url"
+    "time"
+
+    "github.com/go-chi/chi"
+    "github.com/go-kit/kit/log"
+    "github.com/observatorium/observatorium/internal/proxy"
+    "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+    ReadTimeout  = 15 * time.Minute
+    WriteTimeout = time.Minute
+)
+
+type handlerConfiguration struct {
+    logger     log.Logger
+    registry   *prometheus.Registry
+    instrument handlerInstrumenter
+}
+
+// HandlerOption modifies the handler's configuration
+type HandlerOption func(h *handlerConfiguration)
+
+// Logger add a custom logger for the handler to use.
+func Logger(logger log.Logger) HandlerOption {
+    return func(h *handlerConfiguration) {
+        h.logger = logger
+    }
+}
+
+// Registry adds a custom Prometheus registry for the handler to use.
+func Registry(r *prometheus.Registry) HandlerOption {
+    return func(h *handlerConfiguration) {
+        h.registry = r
+    }
+}
+
+// HandlerInstrumenter adds a custom HTTP handler instrument middleware for the handler to use.
+func HandlerInstrumenter(instrumenter handlerInstrumenter) HandlerOption {
+    return func(h *handlerConfiguration) {
+        h.instrument = instrumenter
+    }
+}
+
+type handlerInstrumenter interface {
+    NewHandler(labels prometheus.Labels, handler http.Handler) http.HandlerFunc
+}
+
+type nopInstrumentHandler struct{}
+
+func (n nopInstrumentHandler) NewHandler(labels prometheus.Labels, handler http.Handler) http.HandlerFunc {
+    return handler.ServeHTTP
+}
+
+func NewHandler(read, write *url.URL, opts ...HandlerOption) http.Handler {
+    c := &handlerConfiguration{
+        logger:     log.NewNopLogger(),
+        registry:   prometheus.NewRegistry(),
+        instrument: nopInstrumentHandler{},
+    }
+    for _, o := range opts {
+        o(c)
+    }
+
+    r := chi.NewRouter()
+
+    if read != nil {
+        var proxyRead http.Handler
+        {
+            middlewares := proxy.Middlewares(
+                proxy.MiddlewareSetUpstream(read),
+                proxy.MiddlewareLogger(c.logger),
+                proxy.MiddlewareMetrics(c.registry, prometheus.Labels{"proxy": "logsv1-read"}),
+            )
+
+            proxyRead = &httputil.ReverseProxy{
+                Director: middlewares,
+                ErrorLog: proxy.Logger(c.logger),
+                Transport: &http.Transport{
+                    DialContext: (&net.Dialer{
+                        Timeout: ReadTimeout,
+                    }).DialContext,
+                },
+            }
+        }
+        r.Handle("/loki/api/v1/query", c.instrument.NewHandler(
```

Comment:
from discussion with @metalmatze looks like this is in fact the case. Question: do we want to remove the `/loki` prefix from our API and add it instead proxy config so that paths get rewritten?

Reply:
Do you think of `/logs/api/v1/query` rewritten to `/loki/api/v1/query` internally?

Reply:
maybe simply rewriting `/api/v1/query` to `/loki/api/v1/query` so we match the metrics API. Also, that way we avoid a double `/logs/` in the complete API, i.e. `/api/v1/logs` + `/logs/api/v1/query`